### DISCUSSION TITLE
Reached heap limit Allocation failed

### DISCUSSION THREAD

**Discussion by leandergangso** [tailwindlabs/tailwindcss#13341] [Help]
**What version of Tailwind CSS are you using?**

v3.4.1

**What build tool (or framework if it abstracts the build tool) are you using?**

Running `npx tailwindcss -i ... -o ... --watch`

**What version of Node.js are you using?**

Not using

**What browser are you using?**

Firefox

**What operating system are you using?**

Linux Fedora Workstation

**Reproduction URL**

Not quite sure how to, but it seem to be related to how often/quick i save and make css changes that trigger the rebuild. It also has to do with time, as it only happens after a while of actively developing and changing html/css.

**Describe your issue**

Have a pretty basic setup with html,js,css and just running tailwind with `npx tailwindcss --watch`, then after it has ran for a bit and under normal development behavior, I get the following error:

```bash
Rebuilding...

Done in 714ms.

Rebuilding...

Done in 16ms.

Rebuilding...

Done in 21ms.

<--- Last few GCs --->

[73346:0x5567089594f0]   861811 ms: Scavenge 4057.9 (4141.1) -> 4057.2 (4141.1) MB, 5.4 / 0.0 ms  (average mu = 0.332, current mu = 0.005) allocation failure; 
[73346:0x5567089594f0]   861815 ms: Scavenge 4057.9 (4141.1) -> 4057.4 (4143.6) MB, 4.1 / 0.0 ms  (average mu = 0.332, current mu = 0.005) allocation failure; 
[73346:0x5567089594f0]   862604 ms: Mark-sweep 4059.2 (4143.6) -> 4057.8 (4144.1) MB, 786.9 / 0.0 ms  (average mu = 0.251, current mu = 0.025) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0x7f262d01657c node::Abort() [/lib64/libnode.so.108]
 2: 0x7f262cf01d1b  [/lib64/libnode.so.108]
 3: 0x7f262e31a504 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/lib64/libnode.so.108]
 4: 0x7f262e31a8f8 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/lib64/libnode.so.108]
 5: 0x7f262e4f9309  [/lib64/libnode.so.108]
 6: 0x7f262e5118d9 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/lib64/libnode.so.108]
 7: 0x7f262e4ebc8a v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/lib64/libnode.so.108]
 8: 0x7f262e4ecda8 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/lib64/libnode.so.108]
 9: 0x7f262e4d0335 v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/lib64/libnode.so.108]
10: 0x7f262e8f5b7f v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/lib64/libnode.so.108]
11: 0x7f262e20b279  [/lib64/libnode.so.108]
```

**Comment by thecrypticace** [comment]
Hey so it looks like Node.js is running out of memory. There's a pretty good chance this is related to scanning _a lot_ of content files including `node_modules`. It could also be related to uses of `safelist`. Can you provide your `tailwind.config.js` file?

**Reply by leandergangso** [reply]
interesting, I don't have that many files or lots of packages installed.
perhaps it's improved by removing html,js from content in tailwind config.

ps: templ is a templating engine for golang that I use.

edit:
maybe @iconify-json package is the cause?
if plugins are parsed or whatever each time a change is detected, then i get why this might happen.

tailwind.config
```js
// import colors from 'tailwindcss/colors'
import defaultTheme from 'tailwindcss/defaultTheme';
import formsPlugin from '@tailwindcss/forms';

import { addDynamicIconSelectors } from '@iconify/tailwind';

/** @type {import('tailwindcss').Config} */
export default {
    content: ["./view/**/*.{templ,html,js}"],

    theme: {
        fontFamily: {
            base: ["Inter", ...defaultTheme.fontFamily.sans],
            headline: ["Poppins", ...defaultTheme.fontFamily.sans],
        },

        extend: {
            boxShadow: {
                around: '0px 0px 16px 4px',
                hard: '2px 2px 0px 0px',
                panel: '0px 2px 4px 0',
            }
        },
    },
    plugins: [
        addDynamicIconSelectors(),
        formsPlugin(),
    ],
};
```

package.json
```json
{
	"scripts": {
		"css": "npx tailwindcss -i ./static/input.css -o ./static/output.css --watch",
		"build": "npx tailwindcss -i ./static/input.css -o ./static/output.css --minify"
	},
	"devDependencies": {
		"@iconify-json/lucide": "^1.1.164",
		"@iconify-json/noto": "^1.1.18",
		"@iconify-json/svg-spinners": "^1.1.2",
		"@iconify/tailwind": "^0.1.4",
		"@tailwindcss/forms": "^0.5.7",
		"tailwindcss": "^3.4.1"
	}
}
```

**Reply by leandergangso** [reply]
after some tweaks it seems the issue got resolved by using a more narrow `content` regex. :+1: 

i do however notice that it goes from rebuilding in `15ms` at the start and up to `500ms` after some time developing.
stopping and starting the tailwind watch cmd again brings the rebuild time back down to `15ms`.

edit:
the change did not completely fixed it, but get it less often.

**Reply by thecrypticace** [reply]
I suspect the iconify package _might_ be the culprit here but I'm not certain. I'll have to look into it.

**Reply by xantari** [reply]
See here: https://github.com/tailwindlabs/tailwindcss/issues/13794
